### PR TITLE
resisc45

### LINKED ISSUE #224: Add RESISC-45 Dataset
Need to download and put on s3

### PR DESCRIPTION
**Pull request by davidslater**
fixes #224

### CODE REVIEW

**Comment by davidslater** [comment]
This turned out to be much more complex than I anticipated. However, I now have a decent grasp of how you create a tensorflow dataset.

The resulting code is pretty small, but there are a lot of gotchas.

On the upside, once we have a good pipeline for this, tfds can handle a lot of the annoying download / validate / extract / cleanup stuff.

**Comment by ng390** [comment]
LGTM other than the nit comment. Since this dataset fits in memory, should we add a similar test as for the other in-memory datasets?

File: armory/data/datasets.py [hunk 1]
```diff
@@ -326,12 +344,44 @@ def _read_images(prefix, gtFile, im_list, label_list):
     )
 
 
+def resisc45(
+    preprocessing_fn: Callable = None, dataset_dir: str = None,
+) -> (np.ndarray, np.ndarray, np.ndarray, np.ndarray):
```
[the remark this answers is not in the document]
Comment (davidslater) [reply armory/data/datasets.py]: The type hinting is wrong - I will fix that. Actually, I will completely remove the type hints until that is resolved.

The outputs are waiting for the generators update. Added a "blocked" label.

**Comment by davidslater** [comment]
Use #234 instead.

**Pull request closed without merging**